traitlets/config/tests/test_application.py

```diff
@@ -66,7 +66,7 @@ class MyApp(Application):
 
     aliases = Dict({
                     ('fooi', 'i') : 'Foo.i',
-                    ('j', 'fooj') : 'Foo.j',
+                    ('j', 'fooj') : ('Foo.j', "`j` terse help msg"),
```

Comment:
I have difficulty with that, since I can't think of any cases where I would want the two messages differ. I see the rationale in the OP, but it's not accurate, for instance:

> For flags, it is possible (actually forced) to provide shorter versions

There's a field for a help string only because it's not retrievable from the traits. There is no restriction on its length.

The only difference between `--help` output and `--help-all` output should be whether the entirety of classes are walked, vs only those items with aliases and flags. The `.help` key exists solely for the purpose of populating this output, so I don't understand why it wouldn't be used.